generator/src/test/java/com/graphicsfuzz/generator/tool/GenerateTest.java

```diff
@@ -373,4 +376,24 @@ public void testNoInjectionSwitchIfDisabled() throws Exception {
 
   }
 
+  @Test
+  public void testStructUniform() throws Exception {
+    // Checks that the generator does not fall over when presented with struct uniforms.
+    final String program = "#version 310 es\n"
+        + "struct S { float x; };"
+        + "uniform S myS;"
+        + "uniform struct T { int y; } myT;"
+        + "void main() {}";
+    final String uniforms = "{}";
+    final File json = temporaryFolder.newFile("shader.json");
+    final File frag = temporaryFolder.newFile("shader.frag");
+    FileUtils.writeStringToFile(frag, program, StandardCharsets.UTF_8);
```

Comment:
Could use `ShaderJobFileOperations.writeStringToFile`

Reply:
If we get #418 merged, it moves to using fileOps everywhere in that file. I could then rebase and make this PR follow suit.

Reply:
OK, @paulthomson : I have done a new push that addresses this, in part thanks to the changes in #418.